pkg/router/seller.go

```diff
@@ -8,262 +8,775 @@ import (
+		if err != nil {
+			logger.Error(err)
+			return echo.NewHTTPError(http.StatusInternalServerError, "failed to get coupon detail")
+		}
+		result.Tags, err = pg.Queries.SellerGetCouponTag(c.Request().Context(), db.SellerGetCouponTagParams{SellerName: param.SellerName, CouponID: param.ID})
+		if err != nil {
+			logger.Error(err)
+			return echo.NewHTTPError(http.StatusInternalServerError, "failed to get coupon tags")
+		}
+
+		return c.JSON(http.StatusOK, result)
 	}
 }
 
-// @Summary Seller add coupon
-// @Description Add coupon for shop.
-// @Tags Seller, Shop, Coupon
-// @Accept json
-// @Produce json
-// @Success 200
-// @Failure 401
-// @Router /seller/coupon [post]
+// @Summary		Seller add coupon
+// @Description	Add coupon for shop.
+// @Tags			Seller, Shop, Coupon
+// @Param			type		body	string	true	"Coupon type"	Enums('percentage', 'fixed', 'shipping')
+// @Param			name		body	string	true	"name of coupon"
+// @Param			description	body	string	true	"description of coupon"
+// @Param			discount	body	number	false	"discount perscent"
+// @Param			start_date	body	time	true	"start date"
+// @Param			expire_date	body	time	true	"expire date"
+// @Accept			json
+// @Produce		json
+// @success		200	{object}	db.Coupon
+// @Failure		400	{object}	echo.HTTPError
+// @Failure		500	{object}	echo.HTTPError
+// @Router			/seller/coupon [post]
 func sellerAddCoupon(pg *db.DB, logger *zap.SugaredLogger) echo.HandlerFunc {
 	return func(c echo.Context) error {
+		var username string = "user1"
+
+		var param db.SellerInsertCouponParams
+		if err := c.Bind(&param); err != nil {
+			logger.Error(err)
+			return err
+		}
+		param.SellerName = username
+		coupon, err := pg.Queries.SellerInsertCoupon(c.Request().Context(), param)
+		if err != nil {
+			logger.Error(err)
+			return echo.NewHTTPError(http.StatusInternalServerError, "failed to add coupon")
+		}
+		return c.JSON(http.StatusOK, coupon)
+	}
+}
 
-		return c.NoContent(http.StatusOK)
+// @Summary		Seller add coupon tag
+// @Description	Add tag on coupon
+// @Tags			Seller, Shop, Coupon,Tag
+// @Accept			json
+// @Param			id		path	string	true	"coupon id"
+// @Param			tag_id	body	int		true	"add tag id"
+// @Produce		json
+// @success		200	{object}	db.Coupon
+// @Failure		400	{object}	echo.HTTPError
+// @Failure		500	{object}	echo.HTTPError
+// @Router			/seller/coupon/{id}/tag [post]
+func sellerAddCouponTag(pg *db.DB, logger *zap.SugaredLogger) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		var username string = "user1"
+
+		var param db.SellerInsertCouponTagParams
+		if err := c.Bind(&param); err != nil {
+			logger.Error(err)
+			return err
+		}
+		param.SellerName = username
+		couponTag, err := pg.Queries.SellerInsertCouponTag(c.Request().Context(), param)
+		if err != nil {
+			logger.Error(err)
+			return echo.NewHTTPError(http.StatusInternalServerError, "failed to add coupon tag")
+		}
+		return c.JSON(http.StatusOK, couponTag)
 	}
 }
 
```

Comment:
Should tag conflict check not depend on `UNIQUE` constraint? Add another check instead?

Reply:
Where is the conflict check?